// umt-envs/index.js
/**
 * Variables de entorno
 * @author Franco Barrientos <franco.barrientos@arzov.com>
 */


// Prefijos
const pfx = {
    USR: 'USR#',
    TEAM: 'TEAM#',
    MEM: 'MEM#',
    CHAT: 'CHAT#',
    MATCH: 'MATCH#',
    PATCH: 'PATCH#',
    STAD: 'STAD#',
    COURT: 'COURT#'
}

// Globales
const gbl = {
    GEOHASH_LENGTH: 6,
    DAYS_TO_EXPIRE: 14,
    TEAMS_SCAN_LIMIT: 20,
    MATCHES_SCAN_LIMIT: 20,
    REQUESTS_SCAN_LIMIT: 20,
    DYNAMODB_CONFIG: {apiVersion: '2012-08-10'},
    LAMBDA_CONFIG: {apiVersion: '2015-03-31'}
}

// Defectos
const dft = {
    USER: {
        POSITIONS: [''],
        SKILLS: {ATT: {N: '1'}, SPD: {N: '1'}, TEC: {N: '1'}, TWK: {N: '1'}, FCE: {N: '1'}, DEF: {N: '1'}}
    },
    TEAMMEMBER: {
        POSITION: {
            '5v5': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}},
            '7v7': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}},
            '11v11': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}}
        },
        ROLE: ['Player'],
        NUMBER: '0'
    },
    TEAM: {
        FORMATION: {'5v5': {S: '2-1-1'}, '7v7': {S: '3-2-1'}, '11v11': {S: '4-4-2'}},
        PICTURE: ''
    },
    MATCH: {
        POSITIONS: [''],
        STADIUMGEOHASH: '',
        STADIUMID: '',
        COURTID: '0',
        REQSTAT: {AR: {S: 'A'}, RR: {S: 'P'}},
        ALLOWED_PATCHES: '0'
    },
    STADIUM: {
        ADDRESS: ''
    },
    COURT: {
        MATERIAL: ''
    }
}

// Desarrollo local
const dev = {
    LAMBDA_CONFIG: {
        apiVersion: '2015-03-31',
        region: 'localhost',
        endpoint: 'http://127.0.0.1:3001',
        sslEnabled: false
    },
    LAMBDA_CONTAINER_CONFIG: {
        apiVersion: '2015-03-31',
        region: 'localhost',
        endpoint: 'http://host.docker.internal:3001',
        sslEnabled: false
    },
    DYNAMODB_CONFIG: {
        apiVersion: '2012-08-10',
        endpoint: 'http://arzov:8000',
        accessKeyId: 'xxxx',
        secretAccessKey: 'xxxx',
        region: 'localhost'
    },
    MATCHES_SCAN_LIMIT: 5,
    REQUESTS_SCAN_LIMIT: 5,
    TEAMS_SCAN_LIMIT: 5
}

module.exports.pfx = pfx
module.exports.gbl = gbl
module.exports.dft = dft
module.exports.dev = dev
