// umt-envs/index.js
/**
 * Environment variables
 * @author Franco Barrientos <franco.barrientos@arzov.com>
 */


// Prefix
const pfx = {
    METADATA: 'META#',
    USER: 'USR#',
    TEAM: 'TM#',
    TEAM_CHAT: 'TM_CHAT#',
    TEAM_MEMBER: 'TM_MBR#',
    MATCH: 'MTCH#',
    MATCH_CHAT: 'MTCH_CHAT#',
    MATCH_PATCH: 'MTCH_PCH#',
    STADIUM: 'STAD#',
    COURT: 'CRT#'
}

// Global
const gbl = {
    GEOHASH_LENGTH: 6,
    MATCH_DAYS_TO_EXPIRE: 14,
    TEAMCHAT_DAYS_TO_EXPIRE: 90,
    SCAN_LIMIT: 20,
    DYNAMODB_CONFIG: {apiVersion: '2012-08-10'},
    LAMBDA_CONFIG: {apiVersion: '2015-03-31'}
}

// Default
const dft = {
    USER: {
        POSITIONS: [''],
        SKILLS: {
            ATT: {N: '1'},
            SPD: {N: '1'},
            TEC: {N: '1'},
            TWK: {N: '1'},
            FCE: {N: '1'},
            DEF: {N: '1'}
        }
    },
    TEAM_MEMBER: {
        POSITION: {
            '5v5': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}},
            '7v7': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}},
            '11v11': {M: {P: {S: ''}, X: {N: '-1'}, Y: {N: '-1'}}}
        },
        ROLE: ['Player'],
        NUMBER: '0'
    },
    TEAM: {
        FORMATION: {
            '5v5': {S: '2-1-1'},
            '7v7': {S: '3-2-1'},
            '11v11': {S: '4-4-2'}
        },
        PICTURE: ''
    },
    MATCH: {
        POSITIONS: [''],
        STADIUMGEOHASH: '',
        STADIUMID: '',
        COURTID: '0',
        REQSTAT: {AR: {S: 'A'}, RR: {S: 'P'}},
        PATCHES: {CP: {N: '0'}, NP: {N: '0'}},
    },
    STADIUM: {
        ADDRESS: ''
    }
}

// Development
const dev = {
    LAMBDA_CONFIG: {
        apiVersion: '2015-03-31',
        region: 'localhost',
        endpoint: 'http://172.17.0.1:3001',
        sslEnabled: false
    },
    DYNAMODB_CONFIG: {
        apiVersion: '2012-08-10',
        endpoint: 'http://172.17.0.1:8000',
        accessKeyId: 'xxxx',
        secretAccessKey: 'xxxx',
        region: 'localhost'
    },
    SCAN_LIMIT: 5
}

module.exports.pfx = pfx
module.exports.gbl = gbl
module.exports.dft = dft
module.exports.dev = dev
