// umt-utils/index.js
/**
 * Functions library
 * @author Franco Barrientos <franco.barrientos@arzov.com>
 */


/**
 * Delete white spaces and apply uppercase to name input
 * @param {String} name Input name
 */
const cleanName = (name) => {
    return name.toUpperCase().trim()
}

/**
 * Transform input name to an `id`
 * @param {String} name Input name
 */
const nameToId = (name) => {
    // Drop all whitespaces and transform to lowercase
    return name.toLowerCase().replace(/\s+/g, '')
}

/**
 * Parse match data for output
 * @param {Object} data Match data to parse
 */
 const parseMatchOutput = (data) => {
    return {
        teamId1: data.hashKey.S.split('#')[1],
        teamId2: data.rangeKey.S.split('#')[1],
        patches: JSON.stringify(data.patches.M),
        positions: data.positions.SS,
        matchFilter: data.matchFilter.SS,
        schedule: data.schedule.S,
        reqStat: JSON.stringify(data.reqStat.M),
        stadiumGeohash: data.stadiumGeohash.S,
        stadiumId: data.stadiumId.S,
        courtId: data.courtId.N,
        genderFilter: data.genderFilter.SS,
        ageMinFilter: data.ageMinFilter.N,
        ageMaxFilter: data.ageMaxFilter.N,
        geohash: data.geohash.S,
        coords: JSON.stringify(data.coords.M),
        expireOn: data.expireOn.S,
        createdOn: data.createdOn.S,
    }
}

module.exports.cleanName = cleanName
module.exports.nameToId = nameToId
module.exports.parseMatchOutput = parseMatchOutput
